core/src/main/java/org/jruby/ext/date/RubyDate.java

```diff
@@ -1213,6 +1213,7 @@ private RubyNumeric op_minus_date(ThreadContext context, final RubyDate that) {
 
         RubyNumeric subDiff = subMillisDiff(context, that);
         if ( ! subDiff.isZero() ) { // diff += diff_sub;
+            subDiff = subDiff.convertToRational().op_div(context, RubyFixnum.newFixnum(context.runtime, DAY_MS));  // #5493
```

Comment:
It might be more efficient to replace `RubyFixnum.newFixnum(context.runtime, DAY_MS)` with `DAY_MS(context)` here, but if so, the same change should presumably be made to a lot of other places in this file as well.

Reply:
I think you're right, but that's unrelated to your fix. Feel free to do it in another PR, though! cc @kares